### PR TITLE
[noodlemagazine] Add extrator

### PR DESCRIPTION
**Pull request by trassshhub**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Add extractor for noodlemagazine.com.

### CODE REVIEW

File: yt_dlp/extractor/noodlemagazine.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        title = self._og_search_title(webpage)
+        duration = str_to_int(self._html_search_meta('video:duration', webpage, 'duration', default=None))
+        description = self._og_search_property('description', webpage, default='').replace(' watch online hight quality video', '')
+        tags = self._html_search_meta('video:tag', webpage, default='').split(', ')
+        view_count = str_to_int(self._html_search_meta('ya:ovs:views_total', webpage, default=None))
+        like_count = str_to_int(self._html_search_meta('ya:ovs:likes', webpage, default=None))
+        upload_update = str_or_none(self._html_search_meta('ya:ovs:upload_date', webpage, default='').replace('-', ''))
```
Comment (pukkandan) [inline yt_dlp/extractor/noodlemagazine.py:62]: unified_strdate

File: yt_dlp/extractor/noodlemagazine.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            'id': video_id,
+            'formats': formats,
+            'title': title,
+            'thumbnail': thumbnail,
+            'duration': duration,
+            'description': description,
+            'tags': tags,
+            'view_count': view_count,
+            'like_count': like_count,
+            'upload_update': upload_update
```
Comment (pukkandan) [inline yt_dlp/extractor/noodlemagazine.py:91]:
```suggestion
            'upload_date': upload_update
```

File: yt_dlp/extractor/noodlemagazine.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                    'url': r're:^https://.*\.pvvstream.pro/.*extra=',
+                    'quality': '480',
+                    'ext': 'mp4',
+                },
+                {
+                    'url': r're:^https://.*\.pvvstream.pro/.*extra=',
+                    'quality': '720',
+                    'ext': 'mp4',
+                },
+            ]
```
Comment (pukkandan) [inline yt_dlp/extractor/noodlemagazine.py:49]: net needed in tests

File: yt_dlp/extractor/noodlemagazine.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        title = self._og_search_title(webpage)
+        duration = str_to_int(self._html_search_meta('video:duration', webpage, 'duration', default=None))
+        description = self._og_search_property('description', webpage, default='').replace(' watch online hight quality video', '')
+        tags = self._html_search_meta('video:tag', webpage, default='').split(', ')
+        view_count = str_to_int(self._html_search_meta('ya:ovs:views_total', webpage, default=None))
+        like_count = str_to_int(self._html_search_meta('ya:ovs:likes', webpage, default=None))
+        upload_update = str_or_none(self._html_search_meta('ya:ovs:upload_date', webpage, default='').replace('-', ''))
+
+        # fetch json
+        m = self._html_search_regex(r'/' + video_id + r'\?(?:.*&)?m=([^&"\'\s,]+)', webpage, 'm')
```
Comment (pukkandan) [inline yt_dlp/extractor/noodlemagazine.py:65]:
```suggestion
        m = self._html_search_regex(rf'/{video_id}\?(?:.*&)?m=([^&"\'\s,]+)', webpage, 'm')
```
what's m? Use a better name

File: yt_dlp/extractor/noodlemagazine.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        playlist = 'https://adult.noodlemagazine.com/playlist/%s?m=%s' % (video_id, m)
+        info = self._download_json(playlist, video_id)
```
Comment (pukkandan) [inline yt_dlp/extractor/noodlemagazine.py:67]:
```suggestion
        playlist_info = self._download_json(f'https://adult.noodlemagazine.com/playlist/{video_id}?m={m}', video_id)
```

File: yt_dlp/extractor/noodlemagazine.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        thumbnail = self._og_search_property('image', webpage, default=None) or info.get('image')
+
+        formats = []
+
+        for mobj in info.get('sources'):
+            formats.append({
+                'url': traverse_obj(mobj, 'file'),
+                'quality': traverse_obj(mobj, 'label'),
+                'ext': traverse_obj(mobj, 'type'),
+            })
```
Comment (pukkandan) [inline yt_dlp/extractor/noodlemagazine.py:77]:
```suggestion
        formats = [{
            'url':source.get('file'),
            'quality': source.get('label'),
            'ext': source.get('type'),
        } for source in info.get('sources')]
```

**Review by pukkandan — CHANGES_REQUESTED** [review]
Porn sites must return age_limit

File: yt_dlp/extractor/noodlemagazine.py (hunk 4 above)
Comment (trassshhub) [reply yt_dlp/extractor/noodlemagazine.py]: renamed it to 'key'

File: yt_dlp/extractor/noodlemagazine.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        title = self._og_search_title(webpage)
+        duration = parse_duration(self._html_search_meta('video:duration', webpage, 'duration', default=None))
+        description = self._og_search_property('description', webpage, default='').replace(' watch online hight quality video', '')
+        tags = self._html_search_meta('video:tag', webpage, default='').split(', ')
+        view_count = parse_count(self._html_search_meta('ya:ovs:views_total', webpage, default=None))
+        like_count = parse_count(self._html_search_meta('ya:ovs:likes', webpage, default=None))
+        upload_update = unified_strdate(self._html_search_meta('ya:ovs:upload_date', webpage, default=''))
+
+        # fetch json
+        key = self._html_search_regex(rf'/{video_id}\?(?:.*&)?m=([^&"\'\s,]+)', webpage, 'm')
```
Comment (pukkandan) [inline yt_dlp/extractor/noodlemagazine.py:44]:
```suggestion
        key = self._html_search_regex(rf'/{video_id}\?(?:.*&)?m=([^&"\'\s,]+)', webpage, 'key')
```

File: yt_dlp/extractor/noodlemagazine.py [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+            'id': video_id,
+            'formats': formats,
+            'title': title,
+            'thumbnail': thumbnail,
+            'duration': duration,
+            'description': description,
+            'tags': tags,
+            'view_count': view_count,
+            'like_count': like_count,
+            'upload_date': upload_update,
```
Comment (pukkandan) [inline yt_dlp/extractor/noodlemagazine.py:65]:
```suggestion
        upload_date = unified_strdate(self._html_search_meta('ya:ovs:upload_date', webpage, default=''))

        key = self._html_search_regex(rf'/{video_id}\?(?:.*&)?m=([^&"\'\s,]+)', webpage, 'key')
        playlist_info = self._download_json(f'https://adult.noodlemagazine.com/playlist/{video_id}?m={key}', video_id)
        thumbnail = self._og_search_property('image', webpage, default=None) or playlist_info.get('image')

        formats = [{
            'url': source.get('file'),
            'quality': source.get('label'),
            'ext': source.get('type'),
        } for source in playlist_info.get('sources')]

        self._sort_formats(formats)

        return {
            'id': video_id,
            'formats': formats,
            'title': title,
            'thumbnail': thumbnail,
            'duration': duration,
            'description': description,
            'tags': tags,
            'view_count': view_count,
            'like_count': like_count,
            'upload_date': upload_date,
```

File: yt_dlp/extractor/noodlemagazine.py [hunk 9]
```diff
@@ -0,0 +1,67 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from ..utils import (
+    parse_duration,
+    parse_count,
+    unified_strdate
+)
+from .common import InfoExtractor
+
```
Comment (pukkandan) [inline yt_dlp/extractor/noodlemagazine.py:10]:
```suggestion
from .common import InfoExtractor
from ..utils import (
    parse_duration,
    parse_count,
    unified_strdate
)

```

**Pull request merged**